google/ggrc-core
# GGRC-7898 / GGRC-7899 [SV] Implement SOX 302 status change

# Dependencies

_After review this PR should be targeted into `GGRC-7771-sox-skip-verify` branch._



# Issue description

This PR contains SOX 302 status change flow and integration tests. After all builds pass it should be targeted into common branch `GGRC-7771-sox-skip-verify`.

# Steps to test the changes

1. Prepare Assessment Template with _"SOX 302 assessment workflow"_ enabled and some custom attributes with negative options provided;
2. Create Assessment with template created in previous step and assign some verifiers;
3. Provide answers in assessment's LCAs and hit _complete_ btn. Check that assessment status was changed according to SOX 302 flow.

# Solution description

This PR includes following changes:

* New mixin `WithPutBeforeCommitHandable` is added to `rest_handable` mixins module. This mixin allows to handle `model_put_before_commit` signals.

* Mixin `WithSOX302Flow` now is derived from `WithPutBeforeCommitHandable`. Support for SOX 302 status flow is added here on `model_put_before_commit` signals. Also some additional support methods where added on `CustomAttributeDefinition` model and `CustomAttributable` mixin.

* Tests for SOX 302 status change flow are added.

# Sanity checklist




# PR Review checklist
